### PR TITLE
feat: add asciidoc

### LINKED ISSUE #161: Add AsciiDoc for Documentation?
Should we add Ascii-Doc for Documentation (Auth-Flow, etc...) or is Swagger and standard `Readme` enough for this?

### PR DESCRIPTION
**Pull request by Cahllagerfeld**
closes #161 

Ascii-Doc is available for js with npm now.
No need for the ruby-version any more.

- [x] initialize ascii-doc
- [x] create Action

### CODE REVIEW

**Comment by Cahllagerfeld** [comment]
In the stargate-repo is a action for generating the docs. The Docs are getting hosted on GH-Pages then. Do we have to activate something for usage of this feature within this repo?

**Comment by eddiejaoude** [comment]
> Ascii-Doc is available for js with npm now.

oh nice 🎉 

> The Docs are getting hosted on GH-Pages

I can't remember 🤦‍♂️ . I think it pushes to the `gh-pages` branch, then we might have to turn it on in settings

**Comment by Cahllagerfeld** [comment]
Gh-Pages needs to be enabled in the settings. Unfortunaltey I cant do it, so you have to enable it. I created a branch `gh-pages` which should be the base for the page I guess.

**Comment by Cahllagerfeld** [comment]
I think if Gh-Pages is enabled its mergeable

**Comment by Cahllagerfeld** [comment]
I enabled GH-Pages for the repo. In my opinion everything is set up and can be tested

**Comment by eddiejaoude** [comment]
Looks great 👍 

Screenshot 2021-08-06 at 07 06 29

**Comment by eddiejaoude** [comment]
I think the README is getting deployed, not the `docs` folder?

**Comment by eddiejaoude** [comment]
Ignore me, I think it is because the Action has not run yet 🤦‍♂️

File: .github/workflows/docs.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+      - uses: actions/setup-node@v2
+        with:
+          node-version: '16'
+      - name: install dependencies
+        run: npm ci
+      - name: build docs
+        run: npm run docs
+
+      - name: deploy
+        if: github.ref == 'refs/heads/main'
```
Comment (eddiejaoude) [inline .github/workflows/docs.yml:22]: Maybe to test it, this could changed to `issue-161`?

**Comment by Cahllagerfeld** [comment]
did I do something wrong? Seems like the action didnt run :(

**Comment by eddiejaoude** [comment]
oh it looks fine to me 🤔

**Comment by Cahllagerfeld** [comment]
Is running actions from a PR still possible though, because of the crypto-mining issue Github had?

**Comment by eddiejaoude** [comment]
yep it should still be possible, restrictions are usually on new contributors - I just tried to make a change to the `docs` folder and still no trigger 🤔

**Comment by Cahllagerfeld** [comment]
we should remember setting the branch back to `main` before merging

**Comment by eddiejaoude** [comment]
Everything looks fine, I cannot see why the Action is not running 🤷‍♂️

**Comment by Cahllagerfeld** [comment]
Now something ran but seems like failing

**Comment by Cahllagerfeld** [comment]
Seems like it worked

**Comment by eddiejaoude** [comment]
It deployed 🎉 - I will change the trigger back

Screenshot 2021-08-06 at 07 59 28

**Comment by Cahllagerfeld** [comment]
Thats actually an awesome feature 💥

**Comment by eddiejaoude** [comment]
For some reason it is not triggering on `paths` - let's merge and see if it solves the issue, otherwise we can have it run for any change on `main`, not as efficient but will still be fine

**Review by eddiejaoude — APPROVED** [review]
Great 👍

**Pull request merged**